python/cuml/cuml/cluster/dbscan.pyx

```diff
@@ -221,24 +232,62 @@ class DBSCAN(UniversalBase,
     <http://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html>`_.
     """
 
-    _cpu_estimator_import_path = 'sklearn.cluster.DBSCAN'
     core_sample_indices_ = CumlArrayDescriptor(order="C")
     labels_ = CumlArrayDescriptor(order="C")
 
-    _hyperparam_interop_translator = {
-        "metric": {
-            "manhattan": "NotImplemented",
-            "chebyshev": "NotImplemented",
-            "minkowski": "NotImplemented",
-        },
-        "algorithm": {
-            "auto": "brute",
-            "ball_tree": "NotImplemented",
-            "kd_tree": "NotImplemented",
-        },
-    }
-
-    @device_interop_preparation
+    _cpu_class_path = "sklearn.cluster.DBSCAN"
+
+    @classmethod
+    def _get_param_names(cls):
+        return [
+            *super()._get_param_names(),
+            "eps",
+            "min_samples",
+            "max_mbytes_per_batch",
+            "calc_core_sample_indices",
+            "metric",
+            "algorithm",
+        ]
+
+    @classmethod
+    def _params_from_cpu(cls, model):
+        if callable(model.metric):
+            raise UnsupportedOnGPU
+        elif model.metric not in _SUPPORTED_METRICS:
+            raise UnsupportedOnGPU
+
+        if model.algorithm not in ("auto", "brute"):
+            raise UnsupportedOnGPU
+
+        return {
+            "eps": model.eps,
+            "min_samples": model.min_samples,
+            "metric": model.metric,
+            "algorithm": "brute",
+        }
+
+    def _params_to_cpu(self):
+        return {
+            "eps": self.eps,
+            "min_samples": self.min_samples,
+            "metric": self.metric,
+            "algorithm": "brute",
+        }
+
+    def _attrs_from_cpu(self, model):
+        return {
+            "core_sample_indices_": to_gpu(model.core_sample_indices_, order="C"),
+            "labels_": to_gpu(model.labels_, order="C"),
+            **super()._attrs_from_cpu(model),
+        }
+
+    def _attrs_to_cpu(self, model):
+        return {
+            "core_sample_indices_": to_cpu(self.core_sample_indices_, order="C"),
+            "labels_": to_cpu(self.labels_, order="C"),
+            **super()._attrs_to_cpu(model),
+        }
```

Comment:
Shouldn't we transfer the `n_features_in_` or `shape_fit_` attribute?

Reply:
`n_features_in_` and other common attributes are handled by the base class methods. And `DBSCAN` has no `shape_fit_` attribute.